packages/gatsby-image/src/index.js

```diff
@@ -206,22 +207,25 @@ function generateBase64Sources(imageVariants) {
 }
 
 function generateNoscriptSource({ srcSet, srcSetWebp, media, sizes }, isWebp) {
-  const src = isWebp ? srcSetWebp : srcSet
-  const mediaAttr = media ? `media="${media}" ` : ``
-  const typeAttr = isWebp ? `type='image/webp' ` : ``
-  const sizesAttr = sizes ? `sizes="${sizes}" ` : ``
+  const propsObj = {};
 
-  return `<source ${typeAttr}${mediaAttr}srcset="${src}" ${sizesAttr}/>`
+  propsObj["media"] = media;
+  if(isWebp) propsObj["type"] ='image/webp'
+  propsObj["srcset"] = isWebp ? srcSetWebp : srcSet
+  propsObj["sizes"] = sizes
+
+  return <source {...propsObj}/>
 }
 
 function generateNoscriptSources(imageVariants) {
   return imageVariants
     .map(
-      variant =>
-        (variant.srcSetWebp ? generateNoscriptSource(variant, true) : ``) +
-        generateNoscriptSource(variant)
+      (variant, i) =>
+        (<React.Fragment key={i}>
```

Comment:
Why is the `key={i}` change necessary? Was this to test something? (The word `key` doesn't appear anywhere else in this PR and I'm not sure if this is just an API thing)

Reply:
When generating React elements through iteration, a key prop should be added as per official docs. It should however be uniquely identifiable by including some data related to the current iteration beyond the index, as that kind of defeats the purpose(docs explain why).